src/Plugin/GraphQL/DataProducer/Entity/EntityLoadMultiple.php

```diff
@@ -0,0 +1,160 @@
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('entity_type.manager'),
+      $container->get('entity.repository'),
+      $container->get('graphql.buffer.entity')
+    );
+  }
+
+  /**
+   * EntityLoad constructor.
+   *
+   * @param array $configuration
+   *   The plugin configuration array.
+   * @param string $pluginId
+   *   The plugin id.
+   * @param array $pluginDefinition
+   *   The plugin definition array.
+   * @param \Drupal\Core\Entity\EntityTypeManager $entityTypeManager
+   *   The entity type manager service.
+   * @param \Drupal\Core\Entity\EntityRepositoryInterface $entityRepository
+   *   The entity repository service.
+   * @param \Drupal\graphql\GraphQL\Buffers\EntityBuffer $entityBuffer
+   *   The entity buffer service.
+   *
+   * @codeCoverageIgnore
+   */
+  public function __construct(
+    $configuration,
+    $pluginId,
+    $pluginDefinition,
+    EntityTypeManager $entityTypeManager,
+    EntityRepositoryInterface $entityRepository,
+    EntityBuffer $entityBuffer
+  ) {
+    parent::__construct($configuration, $pluginId, $pluginDefinition);
+    $this->entityTypeManager = $entityTypeManager;
+    $this->entityRepository = $entityRepository;
+    $this->entityBuffer = $entityBuffer;
+  }
+
+  /**
+   * @param $type
+   * @param $ids
+   * @param null $language
+   * @param null $bundles
+   * @param \Drupal\Core\Cache\RefinableCacheableDependencyInterface $metadata
+   *
+   * @return \GraphQL\Deferred
+   */
+  public function resolve($type, $ids, $language = NULL, $bundles = NULL, RefinableCacheableDependencyInterface $metadata) {
+    $resolver = $this->entityBuffer->add($type, $ids);
+
+    return new Deferred(function () use ($type, $ids, $language, $bundles, $resolver, $metadata) {
+      if (!$entities = $resolver()) {
+        // If there is no entity with this id, add the list cache tags so that the
+        // cache entry is purged whenever a new entity of this type is saved.
+        $type = $this->entityTypeManager->getDefinition($type);
+        /** @var \Drupal\Core\Entity\EntityTypeInterface $type */
+        $tags = $type->getListCacheTags();
+        $metadata->addCacheTags($tags);
+        return [];
+      }
+
+      foreach ($entities as $id => $entity) {
+        if (isset($bundles) && !in_array($entities[$id]->bundle(), $bundles)) {
+          // If the entity is not among the allowed bundles, don't return it.
+          $metadata->addCacheableDependency($entities[$id]);
+          unset($entities[$id]);
+        }
+
+        if (isset($language) && $language != $entities[$id]->language()
+            ->getId() && $entities[$id] instanceof TranslatableInterface) {
+          $entities[$id] = $entities[$id]->getTranslation($language);
+        }
+
+        $access = $entity->access('view', NULL, TRUE);
+        if (!$access->isAllowed()) {
+          // Do not return the entity if access is denied.
+          unset($entities[$id]);
```

Comment:
I think we are missing cache metadata here?

Reply:
would same as above address this such as : 

```
$metadata->addCacheableDependency($entities[$id]);
```

Reply:
Yes. Add the $access parameter to the cache dependencies.

Reply:
@fubhy @pmelab updated code above ☝️